Feature/saml

am486dx opened the pull request.

am486dx committed:
LIBITD-2563. Partial SAML implementation.

A partial SAML implementation - not fully working yet.

https://umd-dit.atlassian.net/browse/LIBITD-2563

am486dx committed:
feat: working SAML login and logout; fix URL path

SAML integration is now working, and I fixed the 'new group'
button.

https://umd-dit.atlassian.net/browse/LIBITD-2563

peichman-umd commented on src/ipmanager/urls.py:
```diff
@@ -14,25 +14,30 @@
+    path('saml2/', include('djangosaml2.urls')),
+    path('home', HomeView.as_view(), name='home_page'),
+    path('django-admin', admin.site.urls),
+    path('groups', GroupsView.as_view(), name='groups'),
```
Need to add the trailing slash back to this path (`groups/`) in order to ensure all the tests pass.

peichman-umd commented on src/ipmanager/urls.py:
```diff
@@ -14,25 +14,30 @@
+    path('users/auth/saml/callback',
+         AssertionConsumerServiceView.as_view(), name="saml2_acs"),
+    path('users/auth/saml/slo', LogoutInitView.as_view(), name='saml2_logout'),
+    path('saml2/', include('djangosaml2.urls')),
```
I think it is fine to leave this in here. I would add a comment explaining why we are manually adding the previous routes at different URLs.

peichman-umd reviewed (changes requested):
Looks good. One typo noted inline to get two failing tests back to working, and a comment on the `djangosaml2.urls`.

am486dx committed:
fix: fix URLS so that tests pass.

Also added docs for local development environment.

https://umd-dit.atlassian.net/browse/LIBITD-2563

am486dx committed:
docs: updated docs to point to SAML dev environment setup.

https://umd-dit.atlassian.net/browse/LIBITD-2563

am486dx committed:
docs: minor fix - committed unsaved change

https://umd-dit.atlassian.net/browse/LIBITD-2563
